### PR TITLE
Test #2

### PR DESCRIPTION
**Pull request by palladidrago**

### CODE REVIEW

File: app/services/next_track.rb [hunk 1]
```diff
@@ -44,6 +44,6 @@ def self.perform radio
       end
     end
     # liquidsoap's json parser wants strings
-    { cue_out: cue_out.to_i.to_s, cue_in: 0.to_s, fade_out: 0.to_s, fade_in: 0.to_s, track: track.cdn_url }
+    {  cue_in: 0.to_s, fade_out: 0.to_s, fade_in: 0.to_s,cue_out: cue_out.to_i.to_s, track: track.cdn_url }
```
Comment (mcfiredrill) [inline app/services/next_track.rb:47]: Why was this changed?

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -10,12 +10,12 @@ services:
     ports:
       - "6379:6379"
     command: "redis-server --requirepass ${REDIS_PASSWORD}"
-  postgres:
-    image: "postgres:9.4"
-    ports:
-      - 5432
+  db:
```
Comment (mcfiredrill) [inline docker-compose-dev.yml:13]: Is this change necessary?

File: spec/mailers/account_mailer_spec.rb [hunk 3]
```diff
@@ -16,6 +16,8 @@
   end
 
   it "sends day_after_trial_ended email" do
+    puts "lol"
```
Comment (mcfiredrill) [inline spec/mailers/account_mailer_spec.rb:19]: Could you remove this? :)

File: spec/models/link_spec.rb [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 RSpec.describe Link, type: :model do
   it "determines the font awesome character to use from the url" do
```
Comment (mcfiredrill) [inline spec/models/link_spec.rb:4]: Could you mark this spec as pending for now, the link feature is not being used yet.(Use `xit`)

File: app/services/next_track.rb (hunk 1 above)
Comment (palladidrago) [reply app/services/next_track.rb:47]: The order of the hashmap is important when checking.

File: docker-compose-dev.yml (hunk 2 above)
Comment (palladidrago) [reply docker-compose-dev.yml:13]: Probably not, I will revert that.

File: spec/mailers/account_mailer_spec.rb (hunk 3 above)
Comment (palladidrago) [reply spec/mailers/account_mailer_spec.rb]: Sorry, I missed that.

File: spec/models/link_spec.rb (hunk 4 above)
Comment (palladidrago) [reply spec/models/link_spec.rb]: will do

File: app/services/next_track.rb (hunk 1 above)
Comment (mcfiredrill) [reply app/services/next_track.rb:47]: Where? It shouldn't matter since you always lookup by key.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,6 @@
+# Comment on bounty
+Everything that requires an AWS key doesn't run. Every error related to '''collect_stats_spec''' , is because i don't have ICECAST_STATS_HOST in .env.
+Every other error is a mail error, if you run the test on your machine it should work, since it's looking for user.subscription in the database.
```
Comment (mcfiredrill) [inline README.md:3]: You can just comment on the github pull request, this doesn't need to go in the README.

File: app/forms/signup_form.rb [hunk 6]
```diff
@@ -27,6 +28,7 @@ def attributes= attrs
     attrs.each do |k,v|
       self.send("#{k}=", v)
     end
+    puts radio.name.present?
```
Comment (mcfiredrill) [inline app/forms/signup_form.rb:31]: Please remove all `puts` before committing.

File: spec/services/collect_stats_spec.rb [hunk 7]
```diff
@@ -8,7 +8,7 @@
   it "collects the stats from icecast's xml" do
     radio = FactoryBot.create :radio, name: "datafruits"
     VCR.use_cassette(RSpec.current_example.metadata[:full_description].to_s) do
-      CollectStats.new(radio).perform
+      CollectStats.new(radio).perform
```
Comment (mcfiredrill) [inline spec/services/collect_stats_spec.rb:11]: I decided to mark this spec as pending https://github.com/datafruits/streampusher-api/pull/84
Can you pull and merge master ?

File: README.md (hunk 5 above)
Comment (palladidrago) [reply README.md]: Done

File: app/forms/signup_form.rb (hunk 6 above)
Comment (palladidrago) [reply app/forms/signup_form.rb]: I have checked all of my commits, i am pretty sure there are no more

File: spec/services/collect_stats_spec.rb (hunk 7 above)
Comment (palladidrago) [reply spec/services/collect_stats_spec.rb:11]: Done

File: app/services/next_track.rb (hunk 1 above)
Comment (palladidrago) [reply app/services/next_track.rb:47]: When rspec checks if those hashmaps are equal to the one hardcoded in, in https://github.com/datafruits/streampusher-api/blob/master/spec/services/next_track_spec.rb,

Comment (palladidrago) [reply app/services/next_track.rb]: You're right, i have reversed that.

**Comment by palladidrago** [comment]
There are a couple of issues i have noticed in the ci: 
11) ScheduledShow dst handling updates all recurrences +1 hour for DST
      Failure/Error: expect(r.start_at).to eq start_at

        expected: 2015-01-01 00:00:00.000000000 +0000
             got: 2037-11-05 00:00:00.000000000 +0000

        (compared using ==)

        Diff:
        @@ -1 +1 @@
        -2015-01-01 00:00:00 UTC
        +Thu, 05 Nov 2037 00:00:00 UTC +00:00
I don't understand how it got 2037, the time seems totally random everycheck, sometimes it doesn't even return an error.
The expect is supposed to be today: start_at = DateTime.now.utc ( in spec/models/scheduled_show_spec.rb)
There is another error like this

**Comment by mcfiredrill** [comment]
> I don't understand how it got 2037, the time seems totally random everycheck, sometimes it doesn't even return an error.

Hmmm I would check if Timecop is properly set up, this should make the time static for each test.

**Comment by mcfiredrill** [comment]
I fixed/marked pending a few more specs, could you merge master again when you get a chance. @palladidrago https://github.com/datafruits/streampusher-api/pull/86
